frontend/src/metabase/visualizations/visualizations/SmartScalar/SmartScalar.unit.spec.js

```diff
@@ -226,4 +236,26 @@ describe("SmartScalar", () => {
 
     expect(screen.getByText("810.8k")).toBeInTheDocument();
   });
+
+  it("should display full dates for native queries (issue #38122)", () => {
+    const queryType = "native";
+    const rows = [
+      ["2019-10-01T04:00:00", 100],
+      ["2019-11-01T04:00:00", 810750.54],
+    ];
+    const insights = [
+      {
+        "last-value": 810750.54,
+        "last-change": 80,
+        "previous-value": 100,
+        unit: "month",
+        col: "Count",
+      },
+    ];
+
+    setup(series({ rows, insights, queryType }));
+
+    expect(screen.queryByText("Nov 2019")).not.toBeInTheDocument();
+    expect(screen.getByText("November 1, 2019, 4:00 AM")).toBeInTheDocument();
+  });
```

Comment:
Had to move the test from `compute.unit.spec.js` to `SmartScalar.unit.spec.js`, since the new trend charts functionality does not exist in 48. See corresponding test in master - https://github.com/metabase/metabase/pull/38344/files#diff-5319611e9f4c7a99acd8792fbeb80a7961e52432a545beaad54ab259db37c4b2R2123-R2170